pkg/storage/imagestore/imagestore.go

```diff
@@ -319,7 +294,8 @@ func (is *ImageStore) GetRepositories() ([]string, error) {
 
 		stores = append(stores, rel)
 
-		return nil
+		// skip additional sub dirs
+		return driver.ErrSkipDir
```

Comment:
if it gets on this lines it means that a repo was validated and inserted in store so it will skip repos subdirs like:
blobs/
.uploads/
.sync/
saving one Stat() call per subdir inside Walk()

Reply:
But does this mean that if `folder1` was validated as a repo `folder1/folder2` will no longer be considered a repo?
I think that right now you can have images in both `folder1` and `folder2`. (Need to test to be sure though)